clang/include/clang/Basic/AttrDocs.td

```diff
@@ -2199,7 +2199,7 @@ attribute was applied. This effect is equivalent to annotating restrict on
 **all** kernel pointer arguments in an OpenCL or SPIR-V kernel.
 
 If ``intel::kernel_args_restrict`` is applied to a function called from a device
-kernel, the attribute is ignored and it is not propagated to a kernel.
+kernel, the attribute is not ignored and it is propagated to a kernel.
```

Comment:
I would change it to '... is propagated to the kernel', since it is the same kernel we are talking about.  (i.e. 'the' instead of 'a' in that sentence.)

Here and the other three places below as well.

Reply:
Good catch . Thanks @premanandrao. Fixed now.